### PR TITLE
[main] Update Unattended ISO Build Instruction

### PR DESCRIPTION
**Pull request by henryli001**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?

Our building doc says using core-legacy.json to build unattended ISO, which is incorrect. core-legacy.json and all other imageconfig jsons in Mariner do not set the TargetDisk entry and thus makes the installer believe this is an offline installation and eventually runs into failure when doing mounting. Adding a core-legacy-unattended-hyperv.json to Mariner so that users can use this as an example to build an unattended ISO that's able to work and updating the build and imageconfig doc.

###### Change Log  

- Update building.md
- Update imageconfig.md
- Add core-legacy-unattended-hyperv.json

###### Does this affect the toolchain?  

NO

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-YYYY-XXXX

###### Test Methodology

- Local Image Build

### CODE REVIEW

File: toolkit/docs/building/building.md [hunk 1]
```diff
@@ -210,7 +210,7 @@ To create an unattended ISO installer (no interactive UI) use `UNATTENDED_INSTAL
 
 ```bash
 # Build the standard ISO with unattended installer
```
Comment (christopherco) [inline toolkit/docs/building/building.md:212]: Please extend this comment to clarify that this unattended install configuration expects to run on Gen1 HyperV VMs
```suggestion
# Build the standard ISO with unattended installer that installs onto the default Gen1 HyperV VM
```

File: toolkit/imageconfigs/core-legacy-unattended-hyperv.json [hunk 2]
```diff
@@ -0,0 +1,67 @@
+{
+    "Disks": [
+        {
+            "PartitionTableType": "gpt",
+            "TargetDisk": {
+                "Type": "path",
+                "Value": "/dev/sda"
+            },
+            "Artifacts": [
```
Comment (christopherco) [inline toolkit/imageconfigs/core-legacy-unattended-hyperv.json:9]: Do we need to keep the Artifacts field in this image configuration?

File: toolkit/docs/formats/imageconfig.md [hunk 3]
```diff
@@ -6,6 +6,9 @@ Image configuration consists of two sections - Disks and SystemConfigs - that de
 ## Disks
 Disks entry specifies the disk configuration like its size (for virtual disks), partitions and partition table.
 
+## TargetDisk
+Defines the physical disk, to which Mariner should be installed. This entry has to be specified and the Type field needs to be set to "path" if intending to build an unattended ISO.
```
Comment (christopherco) [inline toolkit/docs/formats/imageconfig.md:10]: Recommend rewording for better clarity
```suggestion
Required when building unattended ISO installer. This field defines the physical disk to which Mariner should be installed. The `Type` field must be set to `path` and the `Value` field must be set to the desired target disk path.
```

**Review by christopherco — CHANGES_REQUESTED** [review]

File: toolkit/imageconfigs/core-legacy-unattended-hyperv.json (hunk 2 above)
Comment (henryli001) [reply toolkit/imageconfigs/core-legacy-unattended-hyperv.json]: Good catch. It's not needed

**Review by christopherco — APPROVED** [review]

**Pull request merged**